src/server/object_services/md_store.js

```diff
@@ -176,19 +176,21 @@ class MDStore {
         return mongo_utils.populate(docs, doc_path, this._objects.col(), fields);
     }
 
-    async find_object_latest(bucket_id, key) {
-        return this._objects.col().findOne({
+    async find_object_latest(bucket_id, key, upload_in_progress = false) {
+        let params = {
             // index fields:
             bucket: bucket_id,
             key,
             version_past: null,
             // partialFilterExpression:
             deleted: null,
-            upload_started: null,
-        }, {
-            hint: 'latest_version_index',
-            sort: { bucket: 1, key: 1, version_past: 1 },
-        });
+        };
+        let proj = { sort: { bucket: 1, key: 1, version_past: 1 } };
+        if (!upload_in_progress) {
+            params.upload_started = null;
+            proj.hint = 'latest_version_index';
+        }
```

Comment:
I don't understand the cases.
I would prefer to separate into two functions, there seems to be more differences than similarities.